### PR TITLE
feat: Implement remember me option in login screen

### LINKED ISSUE #1301: Add option on login screen to allow the user to store login credentials
**Actual Behaviour**

Currently, the user is not allowed to save he credentials for a quicker login into the app.

**Expected Behaviour**

Add a 'Remember Me' checkbox to allow the user to save his credentials for a quicker login into the app.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by atm1504**
Fixes #1301 
Changes:
Users can now save their login credentials using smart lock features (Credential API of google). Credentials are stored only if the user selects the remember me options. If clicked so, it auto fills the fields, next time when the user wants to log in.

Screenshots for the change: 
sma

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -153,5 +153,7 @@ dependencies {
     //Shimmer
     implementation 'com.facebook.shimmer:shimmer:0.4.0'
 
+    //Smart LOck authentication
+    implementation 'com.google.android.gms:play-services-auth:16.0.1'
```
Comment (arundhati24) [inline app/build.gradle]: Mention the version number in config.gradle and then use it here.

Comment (atm1504) [reply app/build.gradle]: okay sure

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -325,6 +325,7 @@
     <string name="content_type_static">Content Type: Static</string>
 
     <string name="custom_server">Custom Server</string>
+    <string name="remember_me">Remember Me </string>
```
Comment (arundhati24) [inline app/src/main/res/values/strings.xml]: Remove the space after 'Me'

**Comment by arundhati24** [comment]
You have placed all the code in the LoginActivity. The code follows MVP. You should not keep any logic or processing in the view. I suggest you read about MVP and update your code accordingly.

**Comment by arundhati24** [comment]
And, please improve your commit messages.
You could have written a commit message like, "feat: Implement Remember Me feature in the login screen"
You need to write **imperative-style** commit messages.

**Comment by atm1504** [comment]
@arundhati24 @iamareebjamal @batbrain7 please review it

**Comment by iamareebjamal** [comment]
Please make an fdroid build and upload here

**Comment by arundhati24** [comment]
> @arundhati24 @iamareebjamal @batbrain7 please review it

Looks like you have not yet worked on the reviews.

**Comment by atm1504** [comment]
> Can you please check how it is implemented in Attendee app. It will not work with f droid build. As @iamareebjamal suggested make f droid build first then work on play store services.

@liveHarshit I tested it with a fdroid build on my mobile and my friends mobile. It worked fine for both of us.

**Comment by liveHarshit** [comment]
> I tested it with a fdroid build on my mobile and my friends mobile. It worked fine for both of us.

https://forum.f-droid.org/t/avoid-google-play-services-while-using-play-apps/1756/3

**Comment by atm1504** [comment]
Apk: 

[app-fdroid-debug.apk.zip](https://github.com/fossasia/susi_android/files/3281945/app-fdroid-debug.apk.zip)

**Comment by atm1504** [comment]
@iamareebjamal @arundhati24 please review it.

**Comment by liveHarshit** [comment]
> Apk:
> 
> [app-fdroid-debug.apk.zip](https://github.com/fossasia/susi_android/files/3281945/app-fdroid-debug.apk.zip)

Okay, deploy it on fdroid and install from there.

**Comment by iamareebjamal** [comment]
fdroid will not accept the app

**Comment by liveHarshit** [comment]
> fdroid will not accept the app

@atm1504 That's my point. Also discussed in - 

> https://forum.f-droid.org/t/avoid-google-play-services-while-using-play-apps/1756/3

**Comment by iamareebjamal** [comment]
Non open-source libraries can only be used in play store variant as mentioned above several times

**Comment by mariobehling** [comment]
Ok, merging this. Please ensure the Fdroid build does not have the requirement.

**Pull request merged**